packages/extension-base/src/utils/localStorageStores/createStoreDefinition.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019-2023 @polkadot/extension-bg authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { z } from 'zod';
+
+export default <Z extends z.ZodType, D extends z.infer<Z>>(schema: Z, defaultValue?: D) => (namespace: string) => {
+  type Content = z.infer<Z>
+  type CurrentContent = Content
+  type NewContent = Content
+
+  const get = async (): Promise<Content> => {
+    try {
+      const { [namespace]: storageContent = defaultValue } = await chrome.storage.local.get([namespace]);
+
+      return schema.parse(storageContent as unknown); // eslint-disable-line @typescript-eslint/no-unsafe-return
+    } catch (e) {
+      console.error(`The content of the "${namespace}" namespace in local storage does not match the schema:`, e);
+
+      if (defaultValue) {
+        return defaultValue; // eslint-disable-line @typescript-eslint/no-unsafe-return
+      } else {
+        throw e;
+      }
+    }
+  };
+
+  const set = async (value: Content) => {
+    await chrome.storage.local.set({ [namespace]: value });
+  };
+
+  const update = async (updater: (currentContent: CurrentContent) => NewContent): Promise<NewContent> => {
+    const currentContent = await get();
+
+    const newContent = updater(currentContent);
+
+    await chrome.storage.local.set({ [namespace]: newContent });
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+    return newContent;
+  };
+
+  return {
+    get,
+    set,
+    update
+  };
+};
```

Comment:
Perhaps:
```suggestion
export default <Content>(schema: z.ZodType<Content>, defaultValue?: Content) => (namespace: string) => {
  const get = async (): Promise<Content> => {
    try {
      const { [namespace]: storageContent = defaultValue } = await chrome.storage.local.get([namespace]);

      return schema.parse(storageContent as unknown);
    } catch (e) {
      console.error(`The content of the "${namespace}" namespace in local storage does not match the schema:`, e);

      if (defaultValue) {
        return defaultValue;
      } else {
        throw e;
      }
    }
  };

  const set = async (value: Content) => {
    await chrome.storage.local.set({ [namespace]: value });
  };

  const update = async (updater: (currentContent: Content) => Content): Promise<Content> => {
    const currentContent = await get();

    const newContent = updater(currentContent);

    await chrome.storage.local.set({ [namespace]: newContent });

    return newContent;
  };

  return {
    get,
    set,
    update
  };
};
```
It requires change to `satisfies` clause:
```
 satisfies { [namespace: string]: (namespace: string) => {
  get(): Promise<unknown>;
  set(value: unknown): unknown;
  update(updater: (value: unknown) => unknown): void;
}};
```
or we could remove it all together?

Reply:
Nice one 👍 

I think `satisfies` plays an important role though, as it guards future devs adding namespaces to this map from doing something illegal - without it it's a slow american _(direct translation)_.

But I found out that passing `any` to the generic works:
![image](https://github.com/Cardinal-Cryptography/aleph-zero-signer/assets/6209244/abf8307c-b95e-4996-9e77-0e64d9384b9e)
_(note this is one of the very rare cases where `any` is actually fine, since it only defines the generic, not a specific value's type)_